src/FEValues/interface_values.jl

```diff
@@ -0,0 +1,293 @@
+# Defines InterfaceValues and common methods
+"""
+    InterfaceValues(grid::AbstractGrid, quad_rule::FaceQuadratureRule, func_interpol_a::Interpolation, [geom_interpol_a::Interpolation], [func_interpol_b::Interpolation], [geom_interpol_b::Interpolation])
+
+An `InterfaceValues` object facilitates the process of evaluating values, averages, jumps and gradients of shape functions
+and function on the interfaces of finite elements.
+
+**Arguments:**
+
+* `quad_rule_a`: an instance of a [`FaceQuadratureRule`](@ref) for element A.
+* `quad_rule_b`: an instance of a [`FaceQuadratureRule`](@ref) for element B.
+* `func_interpol_a`: an instance of an [`Interpolation`](@ref) used to interpolate the approximated function for element A.
+* `func_interpol_b`: an instance of an [`Interpolation`](@ref) used to interpolate the approximated function for element B.
+* `geom_interpol_a`: an optional instance of an [`Interpolation`](@ref) which is used to interpolate the geometry for element A.
+  It uses the default interpolation of the respective [`RefShape`](@ref) by default.
+* `geom_interpol_b`: an optional instance of an [`Interpolation`](@ref) which is used to interpolate the geometry for element B.
+  It uses the default interpolation of the respective [`RefShape`](@ref) by default.
+ 
+**associated methods:**
+
+* [`shape_value_average`](@ref)
+* [`shape_value_jump`](@ref)
+* [`shape_gradient_average`](@ref)
+* [`shape_gradient_jump`](@ref)
+
+**Common methods:**
+
+* [`reinit!`](@ref)
+* [`getnquadpoints`](@ref)
+* [`getdetJdV`](@ref)
+
+* [`shape_value`](@ref)
+* [`shape_gradient`](@ref)
+* [`shape_divergence`](@ref)
+* [`shape_curl`](@ref)
+
+* [`function_value`](@ref)
+* [`function_gradient`](@ref)
+* [`function_symmetric_gradient`](@ref)
+* [`function_divergence`](@ref)
+* [`function_curl`](@ref)
+* [`spatial_coordinate`](@ref)
+"""
+InterfaceValues
+
+struct InterfaceValues{FVA, FVB} <: AbstractValues
+    face_values_a::FVA
+    face_values_b::FVB
+end
```

Comment:
I just had another stupid idea. Why don't we include the transformed quadrature nodes in this struct?

Reply:
You would still need to permute then, right? Or do you wanna store the transformed values for every rotation and every face combination?

Reply:
The idea is similar to what we need to do for the correction in the cell values for e.g. Nedelec elements, i.e. that we store some static elementary transformations and then compute for the current face the transformed value when we reinit. But we can also store all permutations and just query them.